src/rnpkeys/rnpkeys.cpp

```diff
@@ -230,15 +230,27 @@ bool
 rnp_cmd(rnp_cfg_t *cfg, cli_rnp_t *rnp, optdefs_t cmd, const char *f)
 {
     const char *key;
+    std::string fs;
 
     switch (cmd) {
     case CMD_LIST_KEYS:
         if (!f) {
-            f = rnp_cfg_getstr(cfg, CFG_USERID);
+            list ids = NULL;
+            if (rnp_cfg_copylist_str(cfg, &ids, CFG_USERID) && list_length(ids) > 0) {
+                f = (fs = (char*)list_front(ids)).c_str();
+            }
+            list_destroy(&ids);
```

Comment:
As the previous - no need to duplicate list, while you can get the element via some helper function.